modules/ROOT/pages/provisioning-virtualbox.adoc

```diff
@@ -0,0 +1,186 @@
+
+You can now run `coreos-installer` inside the VM:
+[source, bash]
+----
+sudo coreos-installer install /dev/sda \
+    --insecure-ignition --ignition-url http://10.0.2.2:8000/example.ign
+----
+
+TIP: Check out `coreos-installer install --help` for more options on how to install Fedora CoreOS.
+
+Once the installation is complete, you have to remove the live ISO after shutting down (`sudo shutdown -h now`) the VM. Once the ISO is removed, you can start the VM anew. The actual first boot process begins now. It is at this time that Ignition ingests the configuration file and provisions the system as specified.
+
+[source, bash]
+----
+vboxmanage storageattach "${VM_NAME}" --storagectl IDE --type dvddrive --port 1 --device 0 --medium emptydrive
+vboxmanage startvm "${VM_NAME}" --type headless
+# Once the VM is available, you can log in:
+ssh core@localhost -p 2222
+----
+
+== Installing from Raw
+
+To install FCOS into a VirtualBox virtual machine using the Raw disk image, follow these steps:
+
+- Download the latest ISO image from the https://getfedora.org/coreos/download?tab=metal_virtualized&stream=stable[download page] or with podman (see https://coreos.github.io/coreos-installer/cmd/download/[documentation] for options):
+[source, bash]
+----
+podman run --security-opt label=disable --pull=always --rm -v .:/data -w /data \
+    quay.io/coreos/coreos-installer:release download -f iso
+----
+
+Before the installation can begin you have to convert the raw disk image to a VirtualBox compatible one:
+[source, bash]
+----
+FCOS_DISK=$(realpath ~/Downloads/fedora-coreos-34.20210808.3.0-metal.x86_64.raw.xz)
+UNPACKED_DISK=$(realpath ~/Downloads/fedora-coreos-34.20210808.3.0-metal.x86_64.raw)
+
+# Unpack the raw disk image
+unxz "${FCOS_DISK}"
+----
+
+[NOTE]
+====
+If you want to supply the Ignition configuration file URL statically, you can modify the raw disk before converting it to a VirtualBox disk image:
+[source, bash]
+----
+TMP_FCOS_BOOT_VOLUME=$(realpath /tmp/fcos_boot_volume)
+mkdir "${TMP_FCOS_BOOT_VOLUME}"
+
+fdisk -l "${UNPACKED_DISK}"
+# calculate the $OFFSET for the mount operation based on the sector size of the disk and the start point of the boot partion
+sudo mount -o loop,offset="${OFFSET}" fedora-coreos-34.20210808.3.0-metal.x86_64.raw "${TMP_FCOS_BOOT_VOLUME}"
+# the boot config you are looking for is located in /loader/entries/ostree-1-fedora-coreos.conf on the boot partition
+sudo vi "${TMP_FCOS_BOOT_VOLUME}/loader/entries/ostree-1-fedora-coreos.conf"
+# ...
+sudo umount "${TMP_FCOS_BOOT_VOLUME}"
+----
+
+The file contains a line containing the string `ignition.platform.id=metal $ignition_firstboot` – add `ignition.config.url=http://10.0.2.2:8000/example.ign` after the first boot parameter and save it. You can now unmount the disk.
```

Comment:
Maybe split the code block in two and move this line right under the vi command?

Reply:
I split that note up a little, please check whether that improves readability.